mne/preprocessing/_fine_cal.py

```diff
@@ -0,0 +1,391 @@
+        data_norm = _norm(data, axis=1)
+        for ii in range(len(cals)):
+            S_tot = _trans_sss_basis(exp, all_coils)
+            data_model = np.dot(
+                np.dot(S_tot / cals[:, np.newaxis],
+                       _col_norm_pinv(S_tot[good])[0]), data_cal[good])
+            with np.errstate(invalid='raise'):
+                err = _norm(data_model - data, axis=1) / data_norm
+            tot_err = _norm(data_model[good] - data[good]) / _norm(data[good])
+            if k == len(rhos) - 1:
+                # We want all magnetometers to be calibrated
+                cal_idx = ii
+            else:
+                if k == 0:
+                    first_err = tot_err
+                cal_idx = np.where(~done)[0][np.argmax(err[~done])]
+            done[cal_idx] = True
+            if cal_idx in good:
+                old_z = zs[cal_idx].copy()
+                objective = partial(
+                    _cal_sss_target, old_z=old_z, all_coils=all_coils,
+                    cal_idx=cal_idx, data_cal=data_cal, data_ch=data[cal_idx],
+                    good=good, S_tot=S_tot, origin=origin, ext_order=ext_order)
+
+                # Figure out the additive term for z-component
+                zs[cal_idx] = fmin_cobyla(objective, old_z, cons=(),
+                                          rhobeg=rhobeg, rhoend=rhoend,
+                                          disp=False)
+
+                # Do in-place adjustment to all_coils
+                _adjust_coils(zs[cal_idx], old_z, all_coils, cal_idx,
+                              return_subset=False)
+                cals[cal_idx] = 1. / _norm(zs[cal_idx])
+                data_cal[cal_idx] = data[cal_idx] * cals[cal_idx]
+
+                # Report
+                angle = _vector_angle(zs[cal_idx], orig_zs[cal_idx])
+                logger.debug(
+                    u'        Optimized %s | err=%0.5f | ×%0.4f | %0.1f°'
+                    % (info_mag['ch_names'][cal_idx], tot_err,
+                       cals[cal_idx], np.abs(angle)))
+            if k == len(rhos) - 1:
+                last_err = _norm(data_model[good] -
+                                 data[good]) / _norm(data[good])
+    zs *= cals[:, np.newaxis]
+    assert np.allclose(_norm(zs, axis=1), 1.)
+    logger.info('        Fit mismatch %0.5f -> %0.5f'
+                % (first_err, last_err))
+    return zs, cals
+
+
+def _adjust_coils(new_z, old_z, all_coils, cal_idx, return_subset):
+    """Helper to adjust coils"""
+    # Turn NX and NY to the plane determined by NZ
+    old_z = old_z / _norm(old_z)
+    new_z = new_z / _norm(new_z)
+    rot = _find_vector_rotation(old_z, new_z)  # additional coil rotation
+    this_sl = all_coils[5][cal_idx]
+    this_rmag = np.dot(rot, all_coils[0][this_sl].T).T
+    this_cosmag = np.dot(rot, all_coils[1][this_sl].T).T
+    if return_subset:
+        all_coils = (this_rmag, this_cosmag, np.zeros(this_rmag.shape[0], int),
+                     1, [True], {0: this_sl})
```

Comment:
> Setting it to np.array([True]) seemed to fix things.

That's probably the right fix.

> Looks like I need to implement the write_fine_calibration function anyways :)

Yes, I haven't gotten around to it. :) I also haven't refactored it with the existing code to read fine cal info in `maxwell.py`, but that is on purpose. I don't know how long it will take this to get merged, and don't want to have to rebase if those files are changed in `master` in the meantime.